orchagent/flexcounterorch.cpp

```diff
@@ -66,12 +69,22 @@ unordered_map<string, string> flexCounterGroupMap =
 
 FlexCounterOrch::FlexCounterOrch(DBConnector *db, vector<string> &tableNames):
     Orch(db, tableNames),
-    m_flexCounterConfigTable(db, CFG_FLEX_COUNTER_TABLE_NAME),
     m_bufferQueueConfigTable(db, CFG_BUFFER_QUEUE_TABLE_NAME),
     m_bufferPgConfigTable(db, CFG_BUFFER_PG_TABLE_NAME),
     m_deviceMetadataConfigTable(db, CFG_DEVICE_METADATA_TABLE_NAME)
 {
     SWSS_LOG_ENTER();
+    m_delayTimer = new SelectableTimer(timespec{.tv_sec = FLEX_COUNTER_DELAY_SEC, .tv_nsec = 0});
+    if (WarmStart::isWarmStart())
+    {
+        auto executor = new ExecutableTimer(m_delayTimer, this, "FLEX_COUNTER_DELAY");
+        Orch::addExecutor(executor);
+        m_delayTimer->start();
+    }
+    else
+    {
+        m_delayTimerExpired = true;
```

Comment:
We used to delay counter polling for all reboot types, and now it turns that counter polling will start right after APPLY_VIEW for cold reboot. Although we don't have a hard requirement for cold reboot time, there is still a question: could you please share some data about how this change would affect cold reboot time?

Reply:
Mainly due to VS tests expecting FC counters to appear immediately, which would fail without putting some delays. Therefore I limited this change to fast/warm reboot only. I am looking for a way to not delay counters only for VS, then I can have this delay for all reboot types.

Reply:
@stepanblyschak Since the change is limited to warm-reboot now, may I know how much time can be saved in warm-reboot after delaying flex counter?

Reply:
@bingwang-ms No time savings, the delay was limited to fast and warm boot already. No delay if FC are present in CONFIG_DB already in cold boot.